packages/notices/src/store/actions.js

```diff
@@ -38,6 +38,7 @@ export function* createNotice( status = DEFAULT_STATUS, content, options = {} )
 		context = DEFAULT_CONTEXT,
 		id = uniqueId( context ),
 		actions = [],
+		type = 'default',
```

Comment:
What's the distinction between `context` and `type`? This may be an instance where the concept of a `context` was developed prematurely to its usage, but I think the general idea could be the same so far as defining a particular treatment / grouping of notices.

Reply:
I though about `context` as a way to distinguish areas of the UI. For example "editor  notices", "settings notices" where you'd display these notices in different positions of the screen. (For example right now we have block notices for image uploads, these seem to translate more to a different context for me).

I didn't see the "context" as something that would have impact on the design/persistence/usage of the notice.

That said, you're the initial implementor here, so you'd know better.

Reply:
> I though about `context` as a way to distinguish areas of the UI. For example "editor notices", "settings notices" where you'd display these notices in different positions of the screen. (For example right now we have block notices for image uploads, these seem to translate more to a different context for me).
> 
> I didn't see the "context" as something that would have impact on the design/persistence/usage of the notice.

The distinction becomes blurry for me based on how much we consider these things to be separate from "notices".

In my mind, `context` could make sense here if we treated them all as notices, and if NoticeList was either not prescriptive or otherwise flexible in the visual styling of individual notice items. If that were the case (and I think it could be possible?), I'd recommend `context`, then render a `NoticeList` which are styled in a way like proposed here.

If we treat it instead as an entirely separate thing, different components, then `context` might not fit as well, and we need a different taxonomy (`type`) to distinguish between a notice and a snackbar.

I guess it can also depend on whether we think that there may be _contexts_ of _snackbars_ as well as of _notices_. In which case, the dual classification may be necessary.

Reply:
> The distinction becomes blurry for me based on how much we consider these things to be separate from "notices".

To summarize a clarifying direct message conversation between @youknowriad and myself: From my perspective, it wasn't very clear to me in what way these, from a data perspective, were different from any other notice in the system. 

They initially seemed largely a visual treatment, which is what motivated me toward considering these as a "context" in which individual notices could be styled as a variation of the `Notice` component (e.g. `.components-notice.is-snackbar`).

As the conversation continued, initial differences were raised, including the "timing" aspect. It occurred to me whether this is something, again, distinct to "snackbar" notices, or which could be a common feature of _any_ notice.

There was finally outlined that these are quite different in _intent_, which I see now had been summarized previously at https://github.com/WordPress/gutenberg/issues/13905#issuecomment-465182984 . At that point, I see that they are in-fact different enough to warrant distinction on a `type` property. `context` can still serve for contextualizing notices within a type (e.g. global notice list vs. `withNotices` notice lists, https://github.com/WordPress/gutenberg/issues/9442#issuecomment-464436637 ).

We will still need to be careful that top-level properties of a notice (currently `content`, `isDismissible`, `speak`, `actions`) can apply to both the existing form of notice and this snackbar type. `isDismissible` is already questionable in this regard.